### ISSUE TITLE
ArangoError throws when converted to JSON

### ISSUE THREAD

**Issue by neerajyadav** [arangodb/arangojs#632]
If we try to `JSON.stringify()` the `ArangoJsError` it throws cyclic error.
``` log
Converting circular structure to JSON
    at JSON.stringify (<anonymous>)
``` 
This is either due to  `http-request` or `http-response` objects included in the error thrown. One of them have self reference. Many of the logger frameworks uses `JSON.stringify()` method.

**Comment by pluma** [comment]
This issue will be addressed in the upcoming major release.

**Comment by alolis** [comment]
Is this fixed because I am still getting in version 7.2.0 by doing a very simple query:

```
 const query = aql`
      FOR item IN ${db.collection('items')}
        RETURN item
    `;

const items = await db.query(query);

console.log(JSON.stringify(items));
```

Throws

```
Converting circular structure to JSON
    --> starting at object with constructor 'LinkedListItem'
    |     property 'before' -> object with constructor 'LinkedListItem'
    --- property 'behind' closes the circle
    TypeError: Converting circular structure to JSON
        --> starting at object with constructor 'LinkedListItem'
        |     property 'before' -> object with constructor 'LinkedListItem'
        --- property 'behind' closes the circle
        at JSON.stringify (<anonymous>)
```

**Comment by alolis** [comment]
@pluma , I am using version 7.2.0 but unfortunately, I still see the issue by doing the following:

```
const query = aql`
      FOR item IN ${this.db.collection('items')}
        RETURN item
    `;

const cursor = await this.db.collection('items');
const items = await cursor.all();

console.log(JSON.stringify(items));
```

Throws:

```
Converting circular structure to JSON
    --> starting at object with constructor 'LinkedListItem'
    |     property 'before' -> object with constructor 'LinkedListItem'
    --- property 'behind' closes the circle
    TypeError: Converting circular structure to JSON
        --> starting at object with constructor 'LinkedListItem'
        |     property 'before' -> object with constructor 'LinkedListItem'
        --- property 'behind' closes the circle
        at JSON.stringify (<anonymous>)
```

Is it me or do you think there is still an issue here? Thanks for your time.
